src/controllers/userController.js

```diff
@@ -121,6 +122,55 @@ const updateUserPassword = async (req, res) => {
      #swagger.responses[401] = { description: 'Error. Invalid credentials.' }
   */
 };
+
+const uploadUserImage = async (req, res) => {
+  const userId = req.params.id;
+
+  // The `express-fileupload` middleware will add any files that were sent using a multipart/form-data request,
+  // to the req.files property
+  // This property will be an **object** where the keys of that object, are the names/properties of the files in the form-data request
+
+  // Doing some error checking to ensure that we have all the info we expected from the request
+  if (!req.files || Object.keys(req.files).length === 0) {
+    return res.status(400).send('No files were uploaded.');
+  }
+  if (!userId) {
+    return res.status(400).send('No user id was provided.');
+  }
+
+  // fetching the user. Keep in mind, this means that the user must have already been created, before we can upload an image.
+  // Maybe for the /register endpoint, this won't make sense to do this way necessarily
+  const user = await User.findById(userId);
+
+  if (!user) {
+    return res.status(400).send('no user found');
+  }
+
+  // Assuming that the request named the file sent to us, `profile`
+  // The contents of this variable are outlined here: https://www.npmjs.com/package/express-fileupload#:~:text=The%20req.files,the%20uploaded%20file
+  //?? CHECK IN POSTMAN WHICH PROPERTIES HAS REQ.FILES????
+  console.log(req.files);
+
+  const profile = req.files.foo;
```

Comment:
We should probably also do a check for if `profile` is null (if the request forgot to pass in a `foo` property or mistyped it or something) and return a 400 with appropriate error message in that case

Reply:
Any particular reason to make the property `foo`, instead of something more descriptive about the use-case?

Reply:
@akosasante will use **userImage** instead of **foo**// just thought after reading in docs it was obligatory to name it _foo_

Reply:
Didn’t know that, thank you !